PCbuild/get_external.py

```diff
@@ -3,23 +3,47 @@
 import argparse
 import os
 import pathlib
+import random
 import sys
 import time
 import zipfile
 from urllib.request import urlretrieve
 
 
+def retrieve_with_retries(download_location,
+                          output_path,
+                          reporthook,
+                          max_retries = 7,
+                          base_delay = 2.25,
+                          max_jitter = 1.0):
+    """Download a file with exponential backoff retry and save to disk."""
+    for attempt in range(max_retries):
```

Comment:
It's a clever implementation, but possibly way more than we need here? Jitter seems overboard, in particular, as we're not running this in parallel.

Why not just do something like:

```suggestion
    for delay in [1, 2, 5, 10, 15, 30, None]:
```

And then re-raise when `not delay`? Just means there's a few less pieces to worry about going wrong, or having to understand when something does go wrong.

Reply:
Sure! I added jitter because we are *kind of* starting these in parallel across jobs, and I was hoping to reduce the odds of two jobs conflicting. But I'm fine simplifying this until that's proven needed.

Reply:
The jobs will jitter enough due to VM acquisition/startup time, we're not going to improve that here. Plus I highly doubt we're causing enough traffic to github.com from our builds to cause other traffic to fail (at least I hope we're not!)

Reply:
Simplified things a bit, hopefully the new version is OK?

I wanted to keep the maximum number of retries as a parameter as we've already needed to modify the backoff for downloads in `generate_sbom.py` (they were still timing out after 30s, so it was upped to 130s), and I don't want to regenerate the list by hand every time it needs to be modified.